sstables/index_reader.hh

```diff
@@ -335,7 +335,7 @@ std::unique_ptr<clustered_index_cursor> promoted_index::make_cursor(shared_sstab
             get_clustering_values_fixed_lengths(sst->get_serialization_header()));
     }
 
-    if (sst->get_version() >= sstable_version_types::mc && use_binary_search_in_promoted_index) {
+    if (sst->get_version() >= sstable_version_types::mc) {
```

Comment:
Well, the least intrusive change would be to move it to the constructor, not remove it completely. But I think we've proven binary search works and we can retire this parameter.